fastestimator/op/tensorop/augmentation/cutmix_batch.py

```diff
@@ -98,32 +97,42 @@ def _get_patch_coordinates(tensor: Tensor, x: Tensor, y: Tensor,
         """
         _, img_height, img_width = get_image_dims(tensor)
 
-        cut_x = img_width * x
-        cut_y = img_height * y
-        cut_w = img_width * tensor_sqrt(1 - lam)
-        cut_h = img_height * tensor_sqrt(1 - lam)
+        if tf.is_tensor(tensor):
+            ht = cast(img_height, "float32")
+            wd = cast(img_width, "float32")
+        else:
+            ht = img_height
+            wd = img_width
+
+        cut_x = wd * x
+        cut_y = ht * y
+        cut_w = wd * tensor_sqrt(1 - lam)
+        cut_h = ht * tensor_sqrt(1 - lam)
         bbox_x1 = cast(tensor_round(clip_by_value(cut_x - cut_w / 2, min_value=0)), "int32")
-        bbox_x2 = cast(tensor_round(clip_by_value(cut_x + cut_w / 2, max_value=img_width)), "int32")
+        bbox_x2 = cast(tensor_round(clip_by_value(cut_x + cut_w / 2, max_value=wd)), "int32")
         bbox_y1 = cast(tensor_round(clip_by_value(cut_y - cut_h / 2, min_value=0)), "int32")
-        bbox_y2 = cast(tensor_round(clip_by_value(cut_y + cut_h / 2, max_value=img_height)), "int32")
+        bbox_y2 = cast(tensor_round(clip_by_value(cut_y + cut_h / 2, max_value=ht)), "int32")
         return bbox_x1, bbox_x2, bbox_y1, bbox_y2, img_width, img_height
 
-    def forward(self, data: Tensor, state: Dict[str, Any]) -> Tuple[Tensor, Tensor]:
+    def forward(self, data: List[Tensor], state: Dict[str, Any]) -> Tuple[Tensor, Tensor]:
+        x, y = data
```

Comment:
If x and y are now both required, then the type hints for inputs needs to change. a single 'str' is no longer an option